subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformInfo.java

```diff
@@ -200,11 +149,16 @@ protected ResolvedTransformInputs resolveInputs() {
                 } else {
                     failure = new DefaultLenientConfiguration.ArtifactResolveException("artifacts", artifactTransformer.getDisplayName(), "artifact transform", failures);
                 }
-                return new ResolvedTransformInputs(failure);
-            } else {
-                ResolvedArtifactResult artifact = Iterables.getOnlyElement(visitor.getArtifacts());
-                return new ResolvedTransformInputs(ImmutableList.of(artifact.getFile()));
+                this.failure = failure;
+                this.result = Collections.emptyList();
+                return;
             }
+            ResolvedArtifactResult artifact = Iterables.getOnlyElement(visitor.getArtifacts());
+
+            TransformArtifactOperation operation = new TransformArtifactOperation(artifact.getId(), artifact.getFile(), artifactTransformer);
+            buildOperationExecutor.run(operation);
```

Comment:
Should we do the check if the `transform` has a cached result here, too? So we only would emit build operations when the artifact is actually transformed?

Reply:
I don't think we should. We'll revamp the whole caching architecture anyway.

@oehme do you know how much effect this `if` has?

https://github.com/gradle/gradle/blob/3ba5478382886306fc5bd62c956a28701011159e/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformingAsyncArtifactListener.java#L50-L54